eosdis-nasa/earthdata-pub-dashboard
# EDPUB-1370: Add Conversation Attachment Capabilities

# Description

Create ability to add attachments to a conversation note

## Spec

See Ticket: https://bugs.earthdata.nasa.gov/browse/EDPUB-1370

---

## Validation

1. Make sure all merge request checks have passed (CI/CD).
2. Push `EDPUB-1370-add-conversation-attachment-capabilities` branch of dashboard and api to SIT
3. Go to the conversations page of an existing request.
4. Add some text for the new note.
5. Click the paperclip button and select a file or some files to attach.
6. Click the 'Send Reply' button
7. Confirm that the new note text and attachment are displayed.

NOTE 1: The addition of the attachment to the note is an asynchronous operation. Sometimes the timeout refresh is not long enough to wait for this asynchronous operation, so the text will be displayed but not the attachment. For the purpose of this testing, it is fine just to refresh the page and confirm functionality. During the IP sprint, I plan to revisit this functionality and attempt to fix this issue.

NOTE 2: This PR / task does not capture how this attachment might be included within email notification. The primary concern here is how to download the file to an unauthenticated user. Again, this aspect of the task will be revisited during the IP sprint.

---
